src/IO/configuration.cpp

```diff
@@ -360,4 +361,113 @@ namespace config {
 
 }
 
+/*!
+    \class ConfigStringParser
+    \ingroup Configuration
+    \brief Configuration parser to c++ string
+
+    This class implements a configuration parser to absorb/flush directly on 
+    c++ string buffers.
+*/
+
+
+/*!
+    Construct a new parser.
+
+    \param XMLorJSON false to activate XML format, true for JSON format if supported,
+    otherwise fallback automatically to XML.
+*/
+ConfigStringParser::ConfigStringParser(bool XMLorJSON)
```

Comment:
Since this is a new class and there are no compatibility concerns, can we use an enum rather than a bool? It will make the code easy to understand (```ConfigPartser(true)``` vs ```ConfigParser(ConfigParser::FORMAT_JSON)```)